client/blocks/daily-post-button/helper.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Internal Dependencies
+ */
+import config from 'config';
+
+const types = {
+	dp_prompt: 'prompt',
+	dp_photo_challenge: 'photo',
+	dp_discover: 'discover'
+};
+export function isDailyPostChallengeOrPrompt( post ) {
+	return post.site_ID === config( 'daily_post_blog_id' ) && !! types[ post.type ];
```

Comment:
you can bump this out to a module level const. config values don't change after they're set.